### PR TITLE
Move consumer logic

### PR DESCRIPTION
**Pull request by KPostOffice**
## Related Issues and Dependencies
https://github.com/thoth-station/investigator/issues/170
## This introduces a breaking change

- [ ] Yes
- [x] No

## This Pull Request implements

I rearranged the structure of thoth.investigator to be more segmented to allow people to more easily add new message consumers. By the time this is done all consumer logic should be moved here

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,3 +79,125 @@ solver: str
...
+
+These messages get processed to update advise justification metrics (`advise_justification_type_number`)
+
+#### HashMismatchMessage
+
+- [HashMismatchMessage](https://github.com/thoth-station/messaging/blob/master/thoth/messaging/hash_mismatch.py)
+
+This message received contains:
+
+```python
```
Comment (pacospace) [inline README.md:103]: Don't we add component name e service version also?

**Review by pacospace — COMMENTED** [review]
One small comment otherwise lgtm

File: README.md (hunk 1 above)
Comment (KPostOffice) [reply README.md:103]: ++

File: thoth/investigator/metrics.py [hunk 2]
```diff
@@ -29,12 +29,24 @@
...
+)
+exceptions = Counter(
+    "investigator_exceptions",
+    "Number of investigation messages which failed to be processed.",
+    labelnames=["message_type"],
+)
+success = Counter(
+    "investigators_processed",
+    "Number of investigation messages which were successfully processed.",
+    lablenames=["message_type"],
```
Comment (pacospace) [inline thoth/investigator/metrics.py:45]: there is a typo here: `labelnames`

File: README.md [hunk 3]
```diff
@@ -74,8 +78,140 @@ package_name: str
...
+```
+
+You can add metrics as you see fit, but if the metric is not specific only to your messages please move it to
+thoth/investigator/metrics.py and set the proper labels to differentiate between messages.
+
+### Other additions
+
+- `thoth/investigator/<message_name>/__init__.py`, please add the function for parsing messages and all the metrics
+
+- `thoth/investigator/<message_name>/__init__.py`, add the function for parsing the new message
```
[the remark this answers is not in the document]
Comment (KPostOffice) [reply README.md:215]: Indeed ++

**Comment by KPostOffice** [comment]
@pacospace This version is running in test now. I ended up just building from my local dir because /deploy wasn't working

**Comment by pacospace** [comment]
> @pacospace This version is running in test now. I ended up just building from my local dir because /deploy wasn't working

@harshad16

**Pull request merged**